dags/inference/maxtext_inference.py

```diff
@@ -35,14 +36,26 @@
 ) as dag:
   test_name_prefix = "maxtext-inference"
   test_models = {
-      "llama2-7b": "llama2_7b",
-      # "mistral": "test_mistral",
-      # "gemma-2b": "gemma/2b/test_gemma",
-      # "gpt3": "test_gpt3",
+      "llama2-7b": {
+          "per_device_batch_sizes": [1, 2, 4],
+          "checkpoint": "gs://inference-benchmarks/models/llama2-7b/2024-04-25-14-01/param-only-decode-ckpt-maxtext/checkpoints/0/items",
+          "maxtext_logs": "gs://inference-benchmarks/models/llama2-7b/2024-04-25-14-01/",
+          "tokenizer": "tokenizer.llama2",
+      },
+      "llama2-13b": {
+          "per_device_batch_sizes": [1, 2],
+          "checkpoint": "gs://inference-benchmarks/models/llama2-13b/2024-04-25-14-01/param-only-decode-ckpt-maxtext/checkpoints/0/items",
+          "maxtext_logs": "gs://inference-benchmarks/models/llama2-13b/2024-04-25-14-01/",
+          "tokenizer": "tokenizer.llama2",
+      },
+      "gemma-7b": {
+          "per_device_batch_sizes": [1, 2, 4],
+          "checkpoint": "gs://inference-benchmarks/models/gemma-7b/2024-04-25-14-01/param-only-decode-ckpt-maxtext/checkpoints/0/items",
```

Comment:
Will the model checkpoints be updated? Do we need to test the latest model checkpoints?

Reply:
These checkpoints are based on the PyTorch checkpoint found here `gs://maxtext-llama/llama2-7b/meta-ckpt`. (Which I pulled from existing MaxText end-to-end [tests](https://github.com/google/maxtext/blob/main/end_to_end/tpu/test_llama2_7b.sh#L25). My understanding is that this checkpoints were obtained from Meta. Though not sure if it was from https://github.com/meta-llama/llama, or HuggingFace. )

What I did was do a PyTorch->MaxText conversion, and then converted that rolled (scanned) version to an unnrolled (unscanned) version. That became the final checkpoint path that we see in the code now. This is to reduce time doing conversions every time. I also wrote an internal document on how to do this.

The checkpoints can be updated if the model weights change or we want to use a different fine-tuned version. Otherwise, we can keep it. @JoeZijunZhou  On this note, when you did your testing, did you download from https://github.com/meta-llama/llama? Also did you use the [chat or non-chat](https://github.com/meta-llama/llama/blob/main/download.sh#L36-L45) versions?

Reply:
I tested with both `gs://maxtext-llama/llama2-7b/meta-ckpt` and non-chat ckpt from `https://github.com/meta-llama/llama`. Do you know if the source would change or not?

Reply:
Do you mean would the source checkpoints change? Not sure, but even if they do get updated versions, I don't think it should matter, at least on performance.

Reply:
Do you mean if there's a way to always use the latest checkpoint from `gs://maxtext-llama/llama2-7b/meta-ckpt` (which is a copy from Meta) or directly from Meta (via HuggingFace, or https://github.com/meta-llama/llama)? 

If we point our path to `gs://maxtext-llama/llama2-7b/meta-ckpt` (and add back the conversion process), this relies on the assumption that the checkpoint there is maintained to be the most current. Will need to look into the ownership there, and guarantees (if any) on it being the latest.

I think if we source directly from Meta, we will be able to pull the latest checkpoint? But I haven't yet tested E2E feasibility, as it requires some manual steps (permissions, etc.) to even download the checkpoint. So not sure if/how it could be automated. 

In general, do we expect the llama2 checkpoints to be updated frequently from Meta? It may not be ideal, but if we don't expect frequent updates to the original weights, then would using this fixed path be good for now?